src/librustc_middle/mir/interpret/value.rs

```diff
@@ -583,7 +583,7 @@ impl<'tcx, Tag> ScalarMaybeUndef<Tag> {
     pub fn not_undef(self) -> InterpResult<'static, Scalar<Tag>> {
         match self {
             ScalarMaybeUndef::Scalar(scalar) => Ok(scalar),
-            ScalarMaybeUndef::Undef => throw_ub!(InvalidUndefBytes(None)),
+            ScalarMaybeUndef::Undef => throw_ub!(InvalidUndefBytes { ptr: None, size: None }),
```

Comment:
we may at some point want to store the size of the undef in the `Undef` just like `Scalar::Raw` has a `size` field.

Reply:
Though if we do that, we may want to have the `ptr` and `size` `Option`s separate as you have it now ...

Reply:
Not sure if we do, the point of the `Size` in the error is to point at a memory dump and point out the region of uninitialized memory that was accessed. This here is a by-value piece of uninitialized data, so there is nothing to point at.